formats/json-tests/commonTest/src/kotlinx/serialization/json/JsonCoerceInputValuesTest.kt

```diff
@@ -142,4 +152,23 @@ class JsonCoerceInputValuesTest : JsonTestBase() {
             assertEquals(e2.message, e1.message)
         }
     }
+
+    @Test
+    fun testNullableEnumWithoutDefault() {
+        val j = Json(json) { explicitNulls = false }
+        parametrizedTest { mode ->
+            assertEquals(NullableEnumWithoutDefault(null), j.decodeFromString("{}"))
+            assertEquals(NullableEnumWithoutDefault(null), j.decodeFromString("""{"e":"incorrect"}"""))
```

Comment:
I think we should add test with `explicitNulls = true` to be sure that the error is thrown on unknown enum value